core/src/main/java/com/novoda/noplayer/exoplayer/ExoPlayerTwoImpl.java

```diff
@@ -0,0 +1,166 @@
+        facade.addListener(new BufferStateForwarder(getBufferStateListeners()));
+        facade.addListener(new VideoSizeChangedForwarder(getVideoSizeChangedListeners()));
+
+        facade.setBitrateChangedListeners(getBitrateChangedListeners());
+        facade.setInfoListeners(getInfoListeners());
+    }
+
+    @Override
+    public boolean isPlaying() {
+        return facade.getPlayWhenReady();
+    }
+
+    @Override
+    public int getVideoWidth() {
+        return facade.getVideoWidth();
+    }
+
+    @Override
+    public int getVideoHeight() {
+        return facade.getVideoHeight();
+    }
+
+    @Override
+    public VideoPosition getPlayheadPosition() {
+        return VideoPosition.fromMillis(facade.getPlayheadPosition());
+    }
+
+    @Override
+    public VideoDuration getMediaDuration() {
+        return VideoDuration.fromMillis(facade.getMediaDuration());
+    }
+
+    @Override
+    public int getBufferPercentage() {
+        return facade.getBufferedPercentage();
+    }
+
+    @Override
+    public void play() {
+        showContainer();
+        heart.startBeatingHeart();
+        facade.setPlayWhenReady(true);
+        getStateChangedListeners().onVideoPlaying();
+    }
+
+    @Override
+    public void play(VideoPosition position) {
+        seekTo(position);
+        play();
+    }
+
+    @Override
+    public void pause() {
+        facade.setPlayWhenReady(false);
+        getStateChangedListeners().onVideoPaused();
+        if (heart.isBeating()) {
+            heart.stopBeatingHeart();
+            heart.forceBeat();
+        }
+    }
+
+    @Override
+    public void seekTo(VideoPosition position) {
+        facade.seekTo(position.inMillis());
+    }
+
+    @Override
+    public void reset() {
+        // TODO
+    }
+
+    @Override
+    public void stop() {
+        facade.stop();
+    }
+
+    @Override
+    public void release() {
+        getPlayerReleaseListener().onPlayerPreRelease(this);
+        getStateChangedListeners().onVideoReleased();
+        heart.stopBeatingHeart();
+        facade.release();
+        videoContainer.hide();
+    }
+
+    @Override
+    public void loadVideo(Uri uri, ContentType contentType) {
+        prepareForwarder.reset();
+        showContainer();
+        facade.prepare(uri, contentType);
+    }
+
+    @Override
+    public void loadVideoWithTimeout(Uri uri, ContentType contentType, Timeout timeout, LoadTimeoutCallback loadTimeoutCallback) {
+        loadVideo(uri, contentType);
+    }
+
+    @Override
+    public PlayerInformation getPlayerInformation() {
+        return new ExoPlayerInformation();
+    }
+
+    @Override
+    public void attach(PlayerView playerView) {
+        videoContainer = VideoContainer.with(playerView.getContainerView());
+        facade.setPlayer(playerView.simplePlayerView());
+        addVideoSizeChangedListener(playerView.getVideoSizeChangedListener());
+        // TODO : Set SubtitleView
```

Comment:
Three different people putting them in 😬  I'll make these more consistent and with better descriptions.